enginetest/queries/index_query_plans.go

```diff
@@ -513,336 +525,372 @@ var IndexPlanTests = []QueryPlanTest{
+			" │   │   │       │   ├─ comp_index_t0.v1:1\n" +
+			" │   │   │       │   └─ 44 (tinyint)\n" +
+			" │   │   │       └─ (comp_index_t0.v2:2 BETWEEN 21 (tinyint) AND 83 (tinyint))\n" +
 			" │   │   └─ AND\n" +
-			" │   │       ├─ GreaterThanOrEqual\n" +
+			" │   │       ├─ LessThanOrEqual\n" +
 			" │   │       │   ├─ comp_index_t0.v1:1\n" +
-			" │   │       │   └─ 28 (tinyint)\n" +
+			" │   │       │   └─ 72 (tinyint)\n" +
 			" │   │       └─ NOT\n" +
 			" │   │           └─ Eq\n" +
 			" │   │               ├─ comp_index_t0.v2:2\n" +
-			" │   │               └─ 68 (tinyint)\n" +
+			" │   │               └─ 27 (tinyint)\n" +
 			" │   └─ AND\n" +
-			" │       ├─ GreaterThanOrEqual\n" +
+			" │       ├─ Eq\n" +
 			" │       │   ├─ comp_index_t0.v1:1\n" +
-			" │       │   └─ 33 (tinyint)\n" +
-			" │       └─ NOT\n" +
-			" │           └─ Eq\n" +
-			" │               ├─ comp_index_t0.v2:2\n" +
-			" │               └─ 39 (tinyint)\n" +
-			" └─ IndexedTableAccess(comp_index_t0)\n" +
-			"     ├─ index: [comp_index_t0.v1,comp_index_t0.v2]\n" +
-			"     ├─ static: [{[33, 46), (NULL, 39)}, {[33, 46), (39, ∞)}, {[46, ∞), (NULL, ∞)}]\n" +
+			" │       │   └─ 35 (tinyint)\n" +
+			" │       └─ (comp_index_t0.v2:2 BETWEEN 78 (tinyint) AND 89 (tinyint))\n" +
+			" └─ ProcessTable\n" +
 			"     └─ Table\n" +
 			"         ├─ name: comp_index_t0\n" +
 			"         └─ columns: [pk v1 v2]\n" +
 			"",
 	},
 	{
-		Query: `SELECT * FROM comp_index_t0 WHERE ((v1<39 AND v2<10) OR (v1>64 AND v2<=15)) AND (v1>=41);`,
-		ExpectedPlan: "IndexedTableAccess(comp_index_t0)\n" +
-			" ├─ index: [comp_index_t0.v1,comp_index_t0.v2]\n" +
-			" ├─ static: [{(64, ∞), (NULL, 15]}]\n" +
-			" └─ Table\n" +
-			"     ├─ name: comp_index_t0\n" +
-			"     └─ columns: [pk v1 v2]\n" +
-			"",
-	},
-	{
-		Query: `SELECT * FROM comp_index_t0 WHERE (((v1<=91) OR (v1<70 AND v2>=23)) OR (v1>23 AND v2<38));`,
+		Query: `SELECT * FROM comp_index_t0 WHERE (((v1<=16) OR (v1>=77 AND v2>77)) OR (v1>19 AND v2>27));`,
 		ExpectedPlan: "IndexedTableAccess(comp_index_t0)\n" +
 			" ├─ index: [comp_index_t0.v1,comp_index_t0.v2]\n" +
-			" ├─ static: [{(NULL, 91], [NULL, ∞)}, {(91, ∞), (NULL, 38)}]\n" +
+			" ├─ static: [{(NULL, 16], [NULL, ∞)}, {(19, ∞), (27, ∞)}]\n" +
 			" └─ Table\n" +
 			"     ├─ name: comp_index_t0\n" +
 			"     └─ columns: [pk v1 v2]\n" +
 			"",
 	},
 	{
-		Query: `SELECT * FROM comp_index_t0 WHERE ((((v1<>45 AND v2=70) OR (v1 BETWEEN 40 AND 96 AND v2 BETWEEN 48 AND 96)) OR (v1<>87 AND v2<31)) OR (v1<>62 AND v2=51)) AND (v1>=47 AND v2<29);`,
+		Query: `SELECT * FROM comp_index_t0 WHERE ((v1>=46) AND (v1>=28 AND v2<>68) OR (v1>=33 AND v2<>39));`,
 		ExpectedPlan: "IndexedTableAccess(comp_index_t0)\n" +
 			" ├─ index: [comp_index_t0.v1,comp_index_t0.v2]\n" +
-			" ├─ static: [{[47, 87), (NULL, 29)}, {(87, ∞), (NULL, 29)}]\n" +
+			" ├─ static: [{[33, 46), (NULL, 39)}, {[33, 46), (39, ∞)}, {[46, ∞), (NULL, ∞)}]\n" +
 			" └─ Table\n" +
```

Comment:
I'm reading this as `(v1>=46 AND v2<>68) OR (v1>=33 AND v2<>39)`, after pushing the AND into the first OR. I think it checks out?

Reply:
Ah... I see it now, thanks for double checking it. You're correct – this **does** include records where v2=68, but that's valid for the filter because it would match the second half of the `OR`. All these expressions were melting my brain last night! 😉